src/sources/aws_s3/sqs.rs

```diff
@@ -38,29 +37,42 @@ lazy_static! {
 pub(super) struct Config {
     pub(super) queue_url: String,
 
-    #[serde(default = "default_poll_interval_secs")]
-    #[derivative(Default(value = "default_poll_interval_secs()"))]
-    pub(super) poll_secs: u64,
+    // restricted to u32 for safe conversion to i64 later
+    #[serde(default = "default_poll_secs")]
+    #[derivative(Default(value = "default_poll_secs()"))]
+    pub(super) poll_secs: u32,
```

Comment:
We may want to update the doc for this since the meaning has changed slightly:

https://github.com/timberio/vector/blob/c25a4c3f84041b0478d744e4611a56e94d4652f7/docs/reference/components/sources/aws_s3.cue#L85-L94

Reply:
Addressed in c9f6b35.